core/src/Streamly/Internal/Data/MutArray/Type.hs

```diff
@@ -573,8 +573,8 @@ isPinned MutArray{..} = Unboxed.isPinned arrContents
 -- /Pre-release/
 {-# INLINE emptyWithAligned #-}
 newArrayWith, emptyWithAligned :: forall m a. (MonadIO m, Unbox a)
-    => (Int -> Int -> m MutByteArray) -> Int -> Int -> m (MutArray a)
-emptyWithAligned alloc alignSize count = do
+    => (Int -> Int -> IO MutByteArray) -> Int -> Int -> m (MutArray a)
```

Comment:
Should mention the reason for IO callbacks and not lifted callbacks, and also mention the github issue in developer comments at the beginning of the module or where the type is defined.